lib/ansible/modules/cloud/vmware/vmware_host_dns.py

```diff
@@ -0,0 +1,256 @@
+                # ["eu.novartis.net", "nibr.novartis.net", "novartis.net"]
+                results['result'][host.name]['current_search_domains'] = host_dns_config.searchDomain
+                results['result'][host.name]['desired_search_domains'] = search_domains
+                host_dns_config.searchDomain = search_domains
+                changed = True
+
+            if changed:
+                if not self.module.check_mode:
+                    try:
+                        host_network_system.UpdateDnsConfig(host_dns_config)
+                        results['result'][host.name]['changed'] = True
+                        results['result'][host.name]['msg'] = "DNS configured for host '%s'" % host.name
+                    except vim.fault.NotFound:
+                        self.module.fail_json(msg="Virtual network adapter for DHCP doesn't exist on host '%s'" % host.name)
+                    except vmodl.fault.InvalidArgument:
+                        self.module.fail_json(msg="Specified IP addresses are invalid for host '%s'" % host.name)
+                    except vim.fault.HostInDomain:
+                        self.module.fail_json(msg="Cannot change host or domain name for host '%s'. The host is part of "
+                                                  "a Windows domain." % host.name)
+                    except vim.fault.HostConfigFault as config_fault:
+                        self.module.fail_json(msg="Failed to configure DNS for host '%s' due to : %s" %
+                                              (host.name, to_native(config_fault.msg)))
+                    except vmodl.fault.NotSupported as not_supported:
+                        self.module.fail_json(msg="Failed to configure DNS for host '%s' : %s" %
+                                              (host.name, to_native(not_supported.msg)))
+                    except (vmodl.RuntimeFault, vmodl.MethodFault) as runtime_fault:
+                        self.module.fail_json(msg="Failed to configure DNS for host '%s' due to : %s" %
+                                              (host.name, to_native(runtime_fault.msg)))
+                else:
+                    results['result'][host.name]['changed'] = True
+                    results['result'][host.name]['msg'] = "DNS will be configured for host '%s'" % host.name
+            else:
+                results['result'][host.name]['changed'] = False
+                results['result'][host.name]['msg'] = "DNS is already configured for host '%s'" % host.name
+
+            host_change_list.append(changed)
+
+        if any(host_change_list):
+            results['changed'] = True
+        self.module.exit_json(**results)
+
+
+def main():
+    """Main"""
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        host_name=dict(required=False, type='str'),
+        domain=dict(required=True, type='str', aliases=['domain_name']),
+        domainname=dict(required=False, type='str', removed_in_version=2.11),
```

Comment:
Not sure - why to add `removed_in_version` in new module

Reply:
Because this parameter was added for backward compatibility with vmware_dns_config.
We could add it as an alias and keep it forever, but I am not in favor of keeping lots of aliases for the same purpose.

Reply:
Please ask @dagwieers

Reply:
If we are creating a new module then why to add old parameters which we don't want in first place.

Reply:
Because this module is a drop-in replacement to the **vmware_dns_config** module we are deprecating. So rather than removing one module, and introducing a new module, we are replacing the old module with a compatible new module, but we are deprecating some of the original parameters instead. End goal is to not break existing playbooks and get rid of some parameters.

Reply:
@ckotte I guess it would be more clear if the PR summary would be updated with the different changes you are making.

Reply:
Done. @Akasurde Does it make sense? Can I resolve the other conversations?